pkg/scheduler/apis/config/v1alpha1/defaults.go

```diff
@@ -79,10 +79,10 @@ func SetDefaults_KubeSchedulerConfiguration(obj *kubescedulerconfigv1alpha1.Kube
 	}
 
 	if len(obj.LeaderElection.LockObjectNamespace) == 0 {
-		obj.LeaderElection.LockObjectNamespace = kubescedulerconfigv1alpha1.SchedulerDefaultLockObjectNamespace
+		obj.LeaderElection.ResourceNamespace = kubeschedulerconfigv1alpha1.SchedulerDefaultLockObjectNamespace
 	}
 	if len(obj.LeaderElection.LockObjectName) == 0 {
```

Comment:
only default `if len(obj.LeaderElection.LockObjectName) == 0 && len(obj.LeaderElection.ResourceName) == 0`, otherwise this defaulting will make a configuration that only sets ResourceNamespace invalid (both fields will be populated and will differ in value so they cannot be converted)